pymeasure/instruments/attocube/adapters.py

```diff
@@ -0,0 +1,108 @@
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import time
+
+from pymeasure.adapters import TelnetAdapter
+
+
+class AttocubeConsoleAdapter(TelnetAdapter):
+    """ Adapter class for connecting to the Attocube Standard Console. This
+    console is a Telnet prompt with password authentication.
+
+    :param host: host address of the instrument
+    :param port: TCPIP port
+    :param passwd: password required to open the connection
+    :param kwargs: Any valid key-word argument for TelnetAdapter
+    """
+    def __init__(self, host, port, passwd, **kwargs):
+        self.read_termination = '\r\n'
+        self.write_termination = self.read_termination
+        super().__init__(host, port, **kwargs)
+        # clear the initial messages from the controller
+        time.sleep(self.query_delay)
+        super().read()
+        # send password and check authorization
+        self.write(passwd, checkAck=False)
+        time.sleep(self.query_delay)
+        ret = super().read()
+        authmsg = ret.split(self.read_termination)[1]
+        if authmsg != 'Authorization success':
+            raise Exception("Attocube authorization failed ('%s')" % authmsg)
+        # switch console echo off
+        self.ask('echo off')
+
+    def check_acknowledgement(self, reply, msg=""):
+        """ checks the last reply of the instrument to be 'OK', otherwise a
+        ValueError is raised.
+
+        :param reply: last reply string of the instrument
+        :param msg: optional message for the eventual error
+        """
+        if reply != 'OK':
+            if msg == "":  # clear buffer
+                msg = reply
+                super().read()
+            raise ValueError("AttocubeConsoleAdapter: Error after command '%s'"
+                             " with message '%s'" % (self.lastcommand, msg))
+
+    def read(self):
+        """ Reads a reply of the instrument which consists of two or more
+        lines. The first ones are the reply to the command while the last one
+        is 'OK' or 'ERROR' to indicate any problem. In case the reply is not OK
+        a ValueError is raised.
+
+        :returns: String ASCII response of the instrument.
+        """
+        raw = super().read().strip(self.read_termination)
+        lines = raw.split('\n')  # line endings inconsistent '\n', or '\r\n'
+        ret = '\n'.join(line.strip() for line in lines[:-1])
```

Comment:
These last two lines can actually be done more elegantly (as you put it) with `rsplit`:
```
ack, ret = raw.rsplit(sep='\n', maxsplit=1)
```
if I'm not deceived.

Reply:
true, except return value needs to be reversed. I changed it, but left there a verbose comment why sep can not be self.read_termination